src/control-plane/backend/lambda/api/service/reporting.ts

```diff
@@ -135,16 +135,15 @@ export class ReportingService {
   private async _buildFunnelQuickSightDashboard(viewName: string, sql: string, tableVisualViewName: string,
     sqlTable: string, query: any, sheetId: string) {
 
-
-    const newFunnelVisualColumns = funnelVisualColumns;
+    const datasetColumns = JSON.parse(JSON.stringify(funnelVisualColumns));
```

Comment:
Why serialize and then deserialize the object?

Reply:
Could we cover the original issue with a unit test?

Reply:
deepCopy Object